src/mountscript.sh

```diff
@@ -602,6 +652,329 @@ ensure_aznfswatchdog()
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add checkHost option to $stunnel_conf_file!"
+        return 1
+    fi
+
+    # TODO: Change to TLSv1.3 once we have TLSv1.3 version enabled.
+    echo "sslVersion = TLSv1.2" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add sslVersion option to $stunnel_conf_file!"
+        return 1
+    fi
+
+    echo "debug = $DEBUG_LEVEL" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add debug option to $stunnel_conf_file!"
+        return 1
+    fi
+
+    stunnel_log_file="$STUNNELDIR/logs/stunnel_$storageaccount.log"
+    echo "output = $stunnel_log_file" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add log file path to $stunnel_conf_file!"
+        return 1
+    fi
+
+    stunnel_pid_file="$STUNNELDIR/logs/stunnel_$storageaccount.pid"
+    echo "pid = $stunnel_pid_file" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add pid file path to $stunnel_conf_file!"
+        return 1
+    fi
+
+    echo >> $stunnel_conf_file
+
+    echo "[$nfs_host]" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add $nfs_host service/entry name to $stunnel_conf_file!"
+        return 1
+    fi
+
+    echo "client = yes" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to 'client = yes' to $stunnel_conf_file!"
+        return 1
+    fi
+
+    echo "accept = $LOCALHOST:$available_port" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add 'accept' info to $stunnel_conf_file!"
+        return 1
+    fi
+
+    echo "connect = $nfs_host:$CONNECT_PORT" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add 'connect' info to $stunnel_conf_file!"
+        return 1
+    fi
+
+    chattr -f +i $stunnel_conf_file
+}
+
+#
+# Mount nfsv4 files share with TLS encryption.
+#
+tls_nfsv4_files_share_mount()
+{
+    vecho "nfs_dir=[$nfs_dir], mount_point=[$mount_point], options=[$OPTIONS], mount_options=[$MOUNT_OPTIONS]."
+
+    # Note the available port
+    available_port=$(get_next_available_port)
+    if [ $? -ne 0 ]; then
+        eecho "Failed to get the available port for nfsv4 mount."
+        exit 1
+    fi
+
+    vecho "Available Port: $available_port"
+
+    if [ -z "$available_port" ]; then
+        eecho "Running out of ports. Nfsv4 has port range $NFSV4_PORT_RANGE_START to $NFSV4_PORT_RANGE_END. All ports from this range are used by other processes."
+        exit 1
+    fi
+
+    EntryExistinMountMap="true"
+
+    stunnel_conf_file="$STUNNELDIR/stunnel_$storageaccount.conf"
```

Comment:
can be a constant

Reply:
no, cannot be a constant. We are changing the attributes of stunnel_conf_file later using chattr command that needs write permission, so cannot make it constant.